zenodo/modules/utils/cli.py

```diff
@@ -440,3 +445,45 @@ def retry_indexing_failed_openaire_records(eager):
         else:
             openaire_direct_index.delay(
                 str(recid.object_uuid), retry=False)
+
+
+@utils.command('delete_openaire_records')
+@with_appcontext
+def delete_openaire_records():
+    """Deletes half deleted OpenAIRE records."""
+    for key in _iter_openaire_direct_index_keys():
+        try:
+            RECID = key.split("openaire_direct_index:")[1]
+
+            record_data = PersistentIdentifier.query.filter_by(
+                pid_type="recid", pid_value=str(RECID), status=PIDStatus.DELETED
+            ).one()
+            record = ZenodoRecord.get_record(
+                    record_data.object_uuid, with_deleted=True
+                    )
+            if "removal_reason" in record:
+                last_valid_revision = record.revisions[-2]
+
+            original_id = openaire_original_id(
+                last_valid_revision, openaire_type(last_valid_revision)
+            )[1]
+            datasource_id = openaire_datasource_id(last_valid_revision)
+
+            params = {
+                "originalId": original_id, "collectedFromId": datasource_id
+            }
+            req = _openaire_request_factory()
+            res = req.delete(
+                current_app.config["OPENAIRE_API_URL"], params=params
+            )
+            res_beta = None
+            if current_app.config["OPENAIRE_API_URL_BETA"]:
+                res_beta = req.delete(
+                    current_app.config["OPENAIRE_API_URL_BETA"], params=params
```

Comment:
Since this logic already exists in the `openaire_delete` task, you can call directly the task here.